README.md

```diff
@@ -100,3 +100,61 @@ google_compute_instance.app.1:
+
+The `apply` plan will run the `terraform_apply` task against the `localhost` target and optionally return the result of the `terraform_output` task. It accepts several fields:
+
+-   `dir`: (Optional) Path to Terraform project directory. Path is relative to CWD, unless an absolute path is specified.
+-   `state`: (Optional) Path to read and save state. Defaults to `terraform.tfstate`. Path is relative to `dir`
+-   `target`: (Optional) Resource to target. Operation will be limited to this resource and its dependencies. Accepts a single resource string or an array of resources.
+-   `var`: (Optional) Set Terraform variables, expects a hash with key value pairs representing variables and values.
+-   `var_file`: (Optional) Set variables in the Terraform configuration from a file. Path is relative to `dir`.
+-   `return_output`: (Optional) Return the result of the `terraform_output` task (defualts to `false`). 
+
+The `terraform_output` task will return the result of executing `terraform output`. It accepts several fields:
+
+-   `dir`: (Optional) Path to Terraform project directory. Path is relative to CWD, unless an absolute path is specified.
+-   `state`: (Optional) Path to read and save state. Defaults to `terraform.tfstate`. Path is relative to `dir`.
+
+## Destroying resources
+
+The `terraform_destroy` task will destroy resources and return the logs printed to stdout. It accepts several fields:
+
+-   `dir`: (Optional) Path to Terraform project directory. Path is relative to CWD, unless an absolute path is specified.
+-   `state`: (Optional) Path to read and save state. Defaults to `terraform.tfstate`. Path is relative to `dir`.
+-   `target`: (Optional) Resource to target. Operation will be limited to this resource and its dependencies. Accepts a single resource string or an array of resources.
+-   `var`: (Optional) Set Terraform variables, expects a hash with key value pairs representing variables and values.
+-   `var_file`: (Optional) Set variables in the Terraform configuration from a file. Path is relative to `dir`.
+
+The `destroy` plan will run the `terraform_destroy` task against the `localhost` and return it's result. It accepts several fields:
+
+-   `dir`: (Optional) Path to Terraform project directory. Path is relative to CWD, unless an absolute path is specified.
+-   `state`: (Optional) Path to read and save state. Defaults to `terraform.tfstate`. Path is relative to `dir`
+-   `target`: (Optional) Resource to target. Operation will be limited to this resource and its dependencies. Accepts a single resource string or an array of resources.
+-   `var`: (Optional) Set Terraform variables, expects a hash with key value pairs representing variables and values.
+-   `var_file`: (Optional) Set variables in the Terraform configuration from a file. Path is relative to `dir`.
+
+
+### Example
+
+In this example plan, resources are applied and then destroyed during plan execution. 
+
```

Comment:
I think it'd be useful to at least show the `outputs` section of the TF file for this example, if not the whole file

Reply:
Show a terraform manifest?

Reply:
Yeah - mostly I think people will want to see what kind of output is valid for being used as targets in a plan.

Reply:
I don't have an example for that, I'm not sure what you mean by `ouput is valid for being used as targets in a plan`? I think maybe the example might be more clear If I just do:
```
plan example(TargetSpec $targets){
  $apply_result = run_plan('terraform::apply', 'dir' => '/home/cas/working_dir/dynamic-inventory-demo', 'return_output' => true)
run_task('important::stuff', $targets, 'task_var' => $apply_result)

  run_plan('terraform_destroy', 'dir' => '/home/cas/working_dir/dynamic-inventory-demo')
}
```

Reply:
That'd also work, though I still think an example of an `outputs` block and/or the JSON object you get back would be helpful. This is a bit inane since the plugin covers this functionality, but it at least demonstrates the syntax and structure traversal:
```
resource "aws_instance" "app" {
...
  tags = { "string" = "HELLO-${count.index}",
"static" = "fun"
  }
}

output "tags" {
  value = zipmap(aws_instance.app.*.public_dns, aws_instance.app.*.tags)
}
```

```
plan test::tf() {
  $targets = run_plan('terraform::apply', 'dir' => '/home/lucy/githubs/terraform-playground/aws', 'return_output' => true)
  # $targets = {tags => { ... , value => {hostname1 => {static => fun, string => HELLO-0}, hostname2 => {static => fun, string => HELLO-1}}}}
  $targets['tags']['value'].map |$host, $tags| {
    Target.new($host, 'vars' => $tags)
  }

  run_command("hostname", $targets)
}
```